Additions/UIView-KIFAdditions.m

```diff
@@ -249,33 +275,34 @@ - (UIAccessibilityElement *)accessibilityElementMatchingBlock:(BOOL(^)(UIAccessi
                     if ([indexPathsForVisibleRows containsObject:indexPath]) {
                         continue;
                     }
-                    
+
                     @autoreleasepool {
-                        // Get the cell directly from the dataSource because UITableView will only vend visible cells
-                        UITableViewCell *cell = [tableView.dataSource tableView:tableView cellForRowAtIndexPath:indexPath];
-                        
+                        // Scroll to the cell and wait for the animation to complete. Using animations here may not be optimal.
+                        [tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionNone animated:animationEnabled];
+
+                        UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
                         UIAccessibilityElement *element = [cell accessibilityElementMatchingBlock:matchBlock notHidden:NO];
-                        
-                        // Remove the cell from the table view so that it doesn't stick around
-                        [cell removeFromSuperview];
-                        
+
                         // Skip this cell if it isn't the one we're looking for
                         if (!element) {
                             continue;
                         }
                     }
-                    
-                    // Scroll to the cell and wait for the animation to complete
-                    BOOL animationEnabled = [KIFUITestActor testActorAnimationsEnabled];
-                    [tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionNone animated:animationEnabled];
+
                     // Note: using KIFRunLoopRunInModeRelativeToAnimationSpeed here may cause tests to stall
-                    CFTimeInterval delay = animationEnabled ? 0.5 : 0.05;
                     CFRunLoopRunInMode(UIApplicationCurrentRunMode, delay, false);
-                    
-                    // Now try finding the element again
                     return [self accessibilityElementMatchingBlock:matchBlock];
                 }
             }
+            if (firstIndexPath) {
+                [tableView scrollToRowAtIndexPath:firstIndexPath atScrollPosition:UITableViewScrollPositionTop animated:NO];
+            }
+			//if we're in a picker (scrollView), let's make sure we set the position back to how it was last set.
```

Comment:
NIT: tabs here again too

Reply:
Yeah, for some reason, our team uses tabs in all of our projects. Make my community software life hell.